clients/kafka-common/src/main/java/org/eclipse/hono/kafka/client/HonoTopic.java

```diff
@@ -0,0 +1,138 @@
+     */
+    public HonoTopic(final Type type, final String tenantId) {
+        Objects.requireNonNull(type);
+        Objects.requireNonNull(tenantId);
+
+        this.tenantId = tenantId;
+        topicString = type.prefix + tenantId;
+    }
+
+    /**
+     * Creates a topic instance from the string representation.
+     *
+     * @param topicString The string to create a topic from.
+     * @return The topic or {@code null} if the string does not contain a valid Hono topic.
+     */
+    public static HonoTopic fromString(final String topicString) {
+        if (topicString.startsWith(Type.TELEMETRY.prefix)) {
+            return new HonoTopic(Type.TELEMETRY, topicString.substring(Type.TELEMETRY.prefix.length()));
+        } else if (topicString.startsWith(Type.EVENT.prefix)) {
+            return new HonoTopic(Type.EVENT, topicString.substring(Type.EVENT.prefix.length()));
+        } else if (topicString.startsWith(Type.COMMAND.prefix)) {
+            return new HonoTopic(Type.COMMAND, topicString.substring(Type.COMMAND.prefix.length()));
+        } else if (topicString.startsWith(Type.COMMAND_INTERNAL.prefix)) {
+            return new HonoTopic(Type.COMMAND_INTERNAL, topicString.substring(Type.COMMAND_INTERNAL.prefix.length()));
+        } else if (topicString.startsWith(Type.COMMAND_RESPONSE.prefix)) {
+            return new HonoTopic(Type.COMMAND_RESPONSE, topicString.substring(Type.COMMAND_RESPONSE.prefix.length()));
+        }
+        return null;
+    }
+
+    /**
+     * Get the tenantId from the HonoTopic.
+     *
+     * @return The tenantId from the HonoTopic.
+     */
+    public String getTenantId() {
+        return tenantId;
+    }
+
+    /**
+     * Get the separator used for the HonoTopic.
+     *
+     * @return the topic separator.
+     */
+    public static String getSeparator() {
+        return SEPARATOR;
+    }
+
+    /**
+     * Returns the string representation of the topic as used by the Kafka client.
+     *
+     * @return The topic as a string.
+     */
+    @Override
+    public String toString() {
+        return topicString;
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final HonoTopic honoTopic = (HonoTopic) o;
+        return topicString.equals(honoTopic.topicString);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(topicString);
+    }
+
+    /**
+     * The type of a Hono specific Kafka topic.
+     */
+    public enum Type {
+
+        TELEMETRY(NAMESPACE + TelemetryConstants.TELEMETRY_ENDPOINT + SEPARATOR),
+        EVENT(NAMESPACE + EventConstants.EVENT_ENDPOINT + SEPARATOR),
+        COMMAND(NAMESPACE + CommandConstants.COMMAND_ENDPOINT + SEPARATOR),
+        COMMAND_INTERNAL(NAMESPACE + CommandConstants.INTERNAL_COMMAND_ENDPOINT + SEPARATOR),
```

Comment:
I'm not sure Hono's internal topics should be handled in the class `HonoTopic`. Do you see other options?